src/ConfigProvider.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\ServiceManager;
+
+use Laminas\ServiceManager\Command\ConfigDumperCommand;
+use Laminas\ServiceManager\Command\FactoryCreatorCommand;
+use Laminas\ServiceManager\Factory\InvokableFactory;
+use Laminas\ServiceManager\Tool\ConfigDumperFactory;
+use Laminas\ServiceManager\Tool\ConfigDumperInterface;
+use Laminas\ServiceManager\Tool\FactoryCreator;
+use Laminas\ServiceManager\Tool\FactoryCreatorInterface;
+use Symfony\Component\Console\Command\Command;
+
+use function class_exists;
+
+use const PHP_SAPI;
+
+/**
+ * @psalm-import-type ServiceManagerConfigurationType from ConfigInterface
+ */
+final class ConfigProvider
+{
+    /**
+     * @return array{
+     *  dependencies: ServiceManagerConfigurationType,
+     *  ...
+     * }
+     */
+    public function __invoke(): array
+    {
+        return [
+            'dependencies' => $this->getServiceDependencies(),
+            'laminas-cli'  => $this->getLaminasCliDependencies(),
+        ];
+    }
+
+    /**
+     * @return ServiceManagerConfigurationType
+     */
+    public function getServiceDependencies(): array
+    {
+        $factories = [
+            ConfigDumperInterface::class   => ConfigDumperFactory::class,
+            FactoryCreatorInterface::class => static fn (): FactoryCreatorInterface => new FactoryCreator(),
+        ];
+
+        if (class_exists(Command::class)) {
+            $factories += [
+                ConfigDumperCommand::class   => InvokableFactory::class,
+                FactoryCreatorCommand::class => InvokableFactory::class,
+            ];
+        }
+
+        return [
+            'factories' => $factories,
+        ];
+    }
+
+    /**
+     * @return array<string,mixed>
+     */
+    private function getLaminasCliDependencies(): array
+    {
+        if (PHP_SAPI !== 'cli') {
+            return [];
+        }
```

Comment:
@Xerkus 
> Commands introduced here are for development where cache should not be enabled generally speaking…

I would even go one step further: the commands will not work correctly when the cache is active.

> …but this is the kind of approach that should not be used at all.

What do you mean here exactly?

Reply:
I can follow the reasoning here, esp. when it comes to config caches.
But again, we are talking about CLI-only tooling and it also verifies that another dependency has to be available.
Don't you think that this is okayish even tho it can have side-effects?

I mean, its not that we blindly merge code just because s1 says "but in PR XY some1 already did that as well and now I want my stuff in here as well based on runtime stuff".

We might have to carefully check what is being added but I cannot think of any CLI dependency which is provided by the servicemanager which needs to be available on production environments and thus, config cache should be disabled in those environments where the commands are in use anyways.

WDYT?

Reply:
> I mean, its not that we blindly merge code just because s1 says "but in PR XY some1 already did that as well and now I want my stuff in here as well based on runtime stuff".

It is more of "they do it, that is how it is intended to be done" case. I would rather not have to check this at all and not give ideas to people doing third party stuff.

Because this approach in general triggers red flags in my mind I want a compelling reason to have it used rather than just "is okayish".

Reply:
As discussed via Slack, I'll drop the `cli` SAPI check as everything would work without it as well.